Add warning for matches (#1047)
meilisearch/documentation: reference/features/search_parameters.md

The beginning of a matching term within a field is indicated by `start`, and its length by `length`.

::: warning
- `start` and `length` are measured in bytes and not the number of characters. For example, `ü` represents two bytes but one character.

> - the `_matchesInfo` returned by MeiliSearch cannot be used for arrays and objects, only for strings.

Reviewer: ```suggestion
- `matchesInfo` cannot be used with arrays and objects, only strings.
```
